.github/actions/restore-node/action.yml

```diff
@@ -181,18 +181,29 @@ runs:
         mkdir -p node_modules/.cache/agoric
         date > node_modules/.cache/agoric/yarn-built
 
-    - name: git dirty check
-      working-directory: ${{ inputs.path }}
-      shell: bash
-      run: |-
-        set -x
-        # Fail if `git status` reports anything other than the following:
-        # * an untracked endo-sha.txt from above
-        # * work tree files that have been staged without further changes
-        #   (e.g., package.json or yarn.lock) as indicated by the Y position
-        #   in "XY PATH" being a space
-        if [ -n "$(git status --porcelain | grep -vE '^[?][?] endo-sha.txt$|^. '; true)" ]; then
-          git status
-          echo "Unexpected dirty git status" 1>&2
-          exit 1
-        fi
+    # Refs: https://github.com/orgs/community/discussions/45342
+    - name: Validate Git Tree in Root Directory
+      if: inputs.path == '.'
```

Comment:
The file path can be either `.` or `./agoric-sdk`, which makes it difficult to determine which composite action is being used. Previously, we could specify the path using the `working-directory` parameter, but this no longer works properly because we now call a separate composite action.

Another option was to concatenate the path value with `.github/actions/with-post-step`, but it does not work with the `uses` keyword.

Reply:
Oh I totally missed the different `uses` path. I think it's another reason to use an off-the-shelf action for this, no relative path to deal with.